lib/ecmascript.ts

```diff
@@ -1382,31 +1382,36 @@ export function PrepareTemporalFields<
   bag: Partial<Record<FieldKeys, unknown>>,
   fields: Array<FieldKeys>,
   requiredFields: RequiredFields,
+  duplicateBehaviour: 'throw' | 'ignore' = 'throw',
   { emptySourceErrorMessage }: FieldPrepareOptions = { emptySourceErrorMessage: 'no supported properties found' }
 ): PrepareTemporalFieldsReturn<FieldKeys, RequiredFields, Owner<FieldKeys>> {
   const result: Partial<Record<AnyTemporalKey, unknown>> = ObjectCreate(null);
   let any = false;
   fields.sort();
+  let previousProperty = undefined;
   for (const property of fields) {
-    let value = bag[property];
-    if (value !== undefined) {
-      any = true;
-      if (BUILTIN_CASTS.has(property)) {
-        // We just has-checked this map access, so there will definitely be a
-        // value.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        value = BUILTIN_CASTS.get(property)!(value);
-      }
-      result[property] = value;
-    } else if (requiredFields !== 'partial') {
-      // TODO: using .call in this way is not correctly type-checked by tsc.
-      // We might need a type-safe Call wrapper?
-      if (ArrayIncludes.call(requiredFields, property)) {
-        throw new TypeError(`required property '${property}' missing or undefined`);
+    if ((property as string) === 'constructor' || (property as string) === '__proto__') {
```

Comment:
> I'm guessing Justin did this to avoid TS errors complaining that property can never be `constructor` or `__proto__`, because FieldKeys is limited to the known subset of Temporal property keys.

Yep, that's correct. 

I think the cast is OK here. And I'm not inclined to mess with the types in this method, given that `PrepareTemporalFields` was, by a very wide margin, the hardest method to get working with TS in the entire polyfill. I don't want to touch anything more. Holler if this is a problem!